src/Ganeti/HTools/Backend/Luxi.hs

```diff
@@ -259,14 +259,15 @@ parseNode ktg [ name, mtotal, mnode, mfree, dtotal, dfree
 parseNode _ v = fail ("Invalid node query result: " ++ show v)
 
 -- | Parses the cluster tags.
-getClusterData :: JSValue -> Result ([String], IPolicy, String)
+getClusterData :: JSValue -> Result ([String], IPolicy, String, Hypervisor)
 getClusterData (JSObject obj) = do
   let errmsg = "Parsing cluster info"
       obj' = fromJSObject obj
   ctags <- tryFromObj errmsg obj' "tags"
   cpol <- tryFromObj errmsg obj' "ipolicy"
   master <- tryFromObj errmsg obj' "master"
-  return (ctags, cpol, master)
+  hypervisor <- tryFromObj errmsg obj' "default_hypervisor"
```

Comment:
Yes, it will return Bad, this is how monadic sequencing works. Thinking about it, even with the string type it could have returned Bad in case the JSObject "default_hypervisor" is entirely missing. I've tried manually parsing it and returning an undef value (Nothing) in case something goes wrong, but it was a bit quirky compared to other parsing functions in the code. Looking around, most of the parsing requires all fields to be present (there are no optional ones), so I think this shouldn't be a problem.